inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/utils/MessageEventReconciliationUtil.kt

```diff
@@ -108,13 +108,14 @@ internal interface MessageEventReconciliationUtil {
                 // Create an empty list which will contain reconciled events. They will be removed
                 // from the relevantEventsCopy iff Trigger is reconciled.
                 val eventsToBeRemoved = ArrayList<Event>() // NOPMD
-
                 // Reconcile each trigger with all relevant events.
                 for (event in relevantEventsCopy) {
                     if (isTriggerReconciled(trigger, event)) {
                         // Add this event to eventsToBeRemoved list because it can't be used again
                         // to satisfy any more triggers.
-                        if (!event.isPersistentType()) {
+                        if (event.isPersistentType()) {
```

Comment:
Sorry, one more thing: Is it possible to add a unit test for the new behavior?

Reply:
added test cases